src/scvi/external/mrvi/_model.py

```diff
@@ -1557,3 +1583,9 @@ def _construct_design_matrix(
         covariates_require_lfc = jnp.array(covariates_require_lfc)
 
         return Xmat, Xmat_names, covariates_require_lfc, offset_indices
+
+    def update_sample_info(self, adata):
```

Comment:
great idea. can you expand on the docstring with the Adata parameter? since it relies on _scvi_sample, it requires it have setup anndata called on it. Maybe, we can just call validate_anndata in the first line of this function.